inwhale/core/uniform.py

```
@@ -322,9 +323,120 @@ def quantize(self, x):
+    As a result , per-channel quantization reduces quantization error, specially for tensors where different channels have very different values.
+    formulas we will use (the same as per-tensor , but they are applied independently to each channel, than a whole tensor):
+    for a tensor X , channels are indexed by c.
+    x_min = min(X[c])
+    x_max = max(X[c])
+
+    scale[c] = (x_max[c] - x_mi[c]) / (qmax - qmin)
+    zero_point[c] = round(qmin - x_min[c] / scale[c]) , then clamped to the rance [qmin, qmax]
+    qx = clamp(round(x / scale[c] + zero_point[c]), qmin, qmax)  // quantized value of x
+    x' = (q - zero_point[c]) * scale[c] // dequantize
+    """
+    def __init__(self, bits, observer, rounding, axis=0, signed=False):
```

Ruff (0.14.10)

Unused method argument: `observer`

(ARG002)